syft/pkg/python_package_metadata.go

```diff
@@ -21,17 +23,45 @@ type PythonFileRecord struct {
 	Size   string            `json:"size,omitempty"`
 }
 
+type PythonDirectURLOriginInfo struct {
+	URL      string `json:"url"`
+	CommitID string `json:"commitId,omitempty"`
+	VCS      string `json:"vcs,omitempty"`
+}
+
 // PythonPackageMetadata represents all captured data for a python egg or wheel package.
 type PythonPackageMetadata struct {
-	Name                 string             `json:"name" mapstruct:"Name"`
-	Version              string             `json:"version" mapstruct:"Version"`
-	License              string             `json:"license" mapstruct:"License"`
-	Author               string             `json:"author" mapstruct:"Author"`
-	AuthorEmail          string             `json:"authorEmail" mapstruct:"Authoremail"`
-	Platform             string             `json:"platform" mapstruct:"Platform"`
-	Files                []PythonFileRecord `json:"files,omitempty"`
-	SitePackagesRootPath string             `json:"sitePackagesRootPath"`
-	TopLevelPackages     []string           `json:"topLevelPackages,omitempty"`
+	Name                 string                     `json:"name" mapstruct:"Name"`
+	Version              string                     `json:"version" mapstruct:"Version"`
+	License              string                     `json:"license" mapstruct:"License"`
+	Author               string                     `json:"author" mapstruct:"Author"`
+	AuthorEmail          string                     `json:"authorEmail" mapstruct:"Authoremail"`
+	Platform             string                     `json:"platform" mapstruct:"Platform"`
+	Files                []PythonFileRecord         `json:"files,omitempty"`
+	SitePackagesRootPath string                     `json:"sitePackagesRootPath"`
+	TopLevelPackages     []string                   `json:"topLevelPackages,omitempty"`
+	DirectURLOrigin      *PythonDirectURLOriginInfo `json:"directUrlOrigin,omitempty"`
+}
+
+type DirectURLOrigin struct {
+	URL         string      `json:"url"`
```

Comment:
It's interesting that these are tagged as underscore.

I know the specification you linked has them tagged this way so it would not unmarshal correctly if we did it with camel case. 

I think I defer to @wagoodman on this for how we want to keep the Metadata shape consistent

I think we have two options here:
1 - create unmarshal structs that then `1-1` translate over to our `Metadata` struct
2 - custom unmarshal to try both cases?

I know the structs here are correct after reviewing the spec, it's just a matter of getting the tagging and presentation correct now.

Reply:
IIUC these structs are not directly used anywhere except for unmarshalling the input `direct_url.json` files and we flatten them out in metadata when we use `PythonDirectURLOriginInfo` struct which has fields as camelCase.

Reply:
That is absolutely my bad - I thought DirectURLOrigin was being put onto metadata not `DirectURLOriginInfo`